client/src/utils/components/dateFilter/DateFilter.tsx

```diff
@@ -33,21 +33,62 @@ export interface DateFilterProps {
 }
 
 /* eslint-disable no-unused-vars */
-export enum dateFilterTypes {
+export enum DateFilterTypes {
   all = "all",
   custom = "custom",
   last90days = "last90days",
   lastMonth = "lastMonth",
   lastWeek = "lastWeek",
   older = "older" // before 90 days
 }
+/* eslint-enable no-unused-vars */
+
+export function stringToDateFilter(str: string) {
+  switch (str) {
+    case "all": return DateFilterTypes.all;
+    case "custom": return DateFilterTypes.custom;
+    case "last90days": return DateFilterTypes.last90days;
+    case "lastMonth": return DateFilterTypes.lastMonth;
+    case "lastWeek": return DateFilterTypes.lastWeek;
+    case "older": return DateFilterTypes.older;
+    default: return undefined;
+  }
```

Comment:
Is it necessary to repeat all the strings here? I'm wondering if we should use `Object.values` instead, something like the following:

```suggestion
  return Object.values(DateFilterTypes).includes(str as DateFilterTypes) ?
    str as DateFilterTypes :
    undefined;
```